### PR TITLE
Update contribution guide, add code of conduct

### PR DESCRIPTION
**Pull request by shawndwells**

### CODE REVIEW

**Comment by shawndwells** [comment]
* @jshcodes this PR converts the existing code of conduct markdown into an actual webpage, which can be referenced in the future (e.g. will become https://opensource.crowdstrike.com/code-of-conduct/). Modified by adding an escalation path/human name people can reach out to.

* @redhatrises this PR also restructures the `contributing` page a little. Based it off https://opensource.com/life/16/3/contributor-guidelines-template-and-tips

**Comment by jshcodes** [comment]
LGTM

File: docs/pages/contributing.md [hunk 1]
```diff
@@ -1,58 +1,96 @@
 ---
-title: Contributing
-description: This is a documentation page.
+title: Contributing to a Project
```
Comment (redhatrises) [inline docs/pages/contributing.md:2]: Let's keep this as `Contributing` as it is pretty standard across all open source projects.

File: docs/pages/contributing.md [hunk 2]
```diff
@@ -1,58 +1,96 @@
...
+    href: '#how-do-i-make-a-contribution'
+  - text: Where can I go for help?
+    href: '#where-can-i-go-for-help'
+  - text: What does the Code of Conduct mean for me?
+    href: '#what-does-the-code-of-conduct-mean-for-me'
 ---
 
-## Welcome to the CrowdStrike Community
+# Welcome!
+CrowdStrike is still new to open source ourselves, and we recognize contributing to a project is no small feat! The guidance here aspires to help onboard new community members into how CrowdStrike-led projects tend to operate, and by extension, make the contribution process easier.
```
Comment (redhatrises) [inline docs/pages/contributing.md:20]:
```suggestion
Welcome and thank you for your interest in contributing to a CrowdStrike project! We recognize contributing to a project is no small feat! The guidance here aspires to help onboard new community members into how CrowdStrike-led projects tend to operate, and by extension, make the contribution process easier.
```

Let's make the first sentence more welcoming and not about our new-ness as we won't always be new. While tru right now, it is a little odd to make new-ness the first sentence.

File: docs/pages/contributing.md [hunk 3]
```diff
@@ -1,58 +1,96 @@
...
 
-Hello and welcome to the CrowdStike community on GitHub! Any and all contributions are welcome. The community may not always merge or act on something, but that is okay!
-Please keep engaging us.
+# How do I make a contribution?
+Never made an open source contribution before? Wondering how contributions work in CrowdStrike projects? Here's a quick rundown!
 
-Speaking of engaging the community, there are a couple of things to consider:
+1. Find an issue that you are interested in addressing or a feature you would like to add. These are often documented in the project repositories themselves, frequently in the `issues` section.
```
Comment (redhatrises) [inline docs/pages/contributing.md:25]:
```suggestion
1. Find an issue that you are interested in addressing, or a feature you would like to add. These are often documented in the project repositories themselves, frequently in the `issues` section.
```

File: docs/pages/contributing.md [hunk 4]
```diff
@@ -1,58 +1,96 @@
...
 
-## Using Git
+3. Clone the repository to your local machine using ``git clone https://github.com/github-username/repository-name.git``.
 
-### Forking Projects
+    GitHub provides documentation on this process, including screenshots, here:
+[https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository#about-cloning-a-repository](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository#about-cloning-a-repository)
 
-When contributing to projects, 
+4. Create a new branch for your changes. This ensures your modifications can be uniquely identified. A local development branch can be created by running a command similar to:
```
Comment (redhatrises) [inline docs/pages/contributing.md:37]:
```suggestion
4. Create a new branch for your changes. This ensures your modifications can be uniquely identified and can help prevent rebasing and history problems. A local development branch can be created by running a command similar to:
```

File: docs/pages/contributing.md [hunk 5]
```diff
@@ -1,58 +1,96 @@
...
 ---
 
-## Welcome to the CrowdStrike Community
+# Welcome!
+CrowdStrike is still new to open source ourselves, and we recognize contributing to a project is no small feat! The guidance here aspires to help onboard new community members into how CrowdStrike-led projects tend to operate, and by extension, make the contribution process easier.
 
-Hello and welcome to the CrowdStike community on GitHub! Any and all contributions are welcome. The community may not always merge or act on something, but that is okay!
-Please keep engaging us.
+# How do I make a contribution?
+Never made an open source contribution before? Wondering how contributions work in CrowdStrike projects? Here's a quick rundown!
```
Comment (redhatrises) [inline docs/pages/contributing.md:23]:
```suggestion
Never made an open source contribution before? Wondering how contributions work in CrowdStrike projects? Here is a quick rundown!
```

While contractions are part of the english language, IMO it is better not to have them in web content for clarity.

File: docs/pages/contributing.md [hunk 6]
```diff
@@ -1,58 +1,96 @@
...
-1. Once the project is forked to your own account, clone the project to your local system. [Example on cloning a repository](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository#about-cloning-a-repository)
-1. Change directory to the repository folder, and create a new branch by running `git checkout -b my_change` where `my_change` is a short drescription of the change being made. Changes should be made in the new branch vs the `main` branch because you normally want your `main` branch to match with the original CrowdStrike repository `main` branch.
-1. Make the desired changes in the repository.
-1. Push the branch back to your GitHub account. For example, `git push origin my_change` where `my_change` is name of the branch.
-1. Create a pull request in the original CrowdStrike project that you forked. [Example on how to create a pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)
+    ``git checkout -b BRANCH-NAME-HERE``
 
-### Rebasing
+5. Make the appropriate changes for the issue you are trying to address or the feature you'd like to add.
```
Comment (redhatrises) [inline docs/pages/contributing.md:41]:
```suggestion
5. Make the appropriate changes for the issue you are trying to address or the feature you would like to add.
```

File: docs/pages/contributing.md [hunk 7]
```diff
@@ -1,58 +1,96 @@
...
 
-You should rarely develop or make changes to your local `main` branch, but sometimes, your `main` branch gets out of date. Also, some pull requests may have been merged that might require
-that the branch on which you are developing contains the latest commits.
+6. Add the file contents of the changed files to the "snapshot" git uses to manage the state of the project (also known as the index). Here's the git command that will add your changes:
```
Comment (redhatrises) [inline docs/pages/contributing.md:43]:
```suggestion
6. Add the file contents of the changed files to the "snapshot" git uses to manage the state of the project (also known as the index). Here is the git command that will add your changes:
```

File: docs/pages/contributing.md [hunk 8]
```diff
@@ -1,58 +1,96 @@
...
 
-#### Squashing
+7. Use `git commit` to store the contents of the index with a descriptive message. This message should outline what was changed. For example:
 
-Before submitting pull requests, please make sure to squash commits in your branch when:
+    ``git commit -m "Added Dockerfile for Ubuntu-based deployments"``
 
-1. You create multiple commits (use `git log` to show history) that have the same commit message like `Update Readme.md`.
-1. Created a bunch of junk or throw away commit messages like `Fix`, `Test`, `Fixed Typos`, etc.
+8. Push your local changes back to gihub.com:
```
Comment (redhatrises) [inline docs/pages/contributing.md:51]:
```suggestion
8. Push your local changes back to your account on github.com:
```

File: docs/pages/contributing.md [hunk 9]
```diff
@@ -1,58 +1,96 @@
...
-1. If you do not see a project, repository, or would like us to consider working on a specific piece of technology, please [open a community ticket](https://github.com/CrowdStrike/community/issues/new). These type of requests should not be made in existing projects.
-1. If there is a specific issue or enhancement you would like to see addressed in a particular project, please open an issue in that project. It might be tempting to open a discussion, but an issue is more appropriate to address the problem.
-1. If you have general questions of a project, please open a discussion in that project instead of a community discussion.
+2. Fork the repository associated with project to your GitHub account. This means that you will have a copy of the repository under *your-GitHub-username/repository-name*. 
 
-Thank you and happy coding!
+   Guidance on how to fork a repository can be found here:
+[https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#fork-an-example-repository](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#fork-an-example-repository).
```
Comment (redhatrises) [inline docs/pages/contributing.md:30]:
```suggestion
   Guidance on how to fork a repository can be found at [https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#fork-an-example-repository](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#fork-an-example-repository).
```

File: docs/pages/contributing.md [hunk 10]
```diff
@@ -1,58 +1,96 @@
...
-1. You create multiple commits (use `git log` to show history) that have the same commit message like `Update Readme.md`.
-1. Created a bunch of junk or throw away commit messages like `Fix`, `Test`, `Fixed Typos`, etc.
+8. Push your local changes back to gihub.com:
 
-[Example on squashing commits using git rebase](https://garrytrinder.github.io/2020/03/squashing-commits-using-interactive-rebase)
+    ``git push origin BRANCH-NAME-HERE``
+
+9. Submit a pull request to the upstream project. Documentation on this process, including screen shots, can be found here:
+
+[https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)
```
Comment (redhatrises) [inline docs/pages/contributing.md:57]:
```suggestion
9. Submit a pull request to the upstream project. Documentation on this process, including screen shots, can be found at [https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)
```

File: docs/pages/contributing.md [hunk 11]
```diff
@@ -1,58 +1,96 @@
...
+
+10. Once submitted, a maintainer will review your pull request. They may ask for additional changes, or clarification, so keep an eye out for communication! GitHub automatically sends an email to your email address whenever someone comments on your pull request.
+
+11. Celebrate your contribution after your pull request is merged! All changes move the project forward, and we thank you for helping the community!
```
Comment (redhatrises) [inline docs/pages/contributing.md:61]:
```suggestion
11. While not all pull requests may be merged, celebrate your contribution whether or not your pull request is merged! All changes move the project forward, and we thank you for helping the community!
```

I am not sure how to say this better, but I feel like we need to acknowledge in some way that some PRs might not get merged.

File: docs/pages/contributing.md [hunk 12]
```diff
@@ -1,58 +1,96 @@
...
+
+## Rebase Early, Rebase Often!
+Projects tend to move at a fast pace, which means your fork may become behind upstream. Keeping your local fork in sync with upstream is called `rebasing`. This ensures your local copy is frequently refreshed with the latest changes from the community. 
+
+Frequenty rebasing is *strongly* encouraged. If your local copy falls to far behind, you may encounter merge conflicts when submitting pull request. If this happens, you'll have to triage (often by hand!) the differents in your local repository versus the changes upstream.
```
Comment (redhatrises) [inline docs/pages/contributing.md:66]:
```suggestion
Frequenty rebasing is *strongly* encouraged. If your local copy falls to far behind, you may encounter merge conflicts when submitting pull request. If this happens, you will have to triage (often by hand!) the differences in your local repository versus the changes upstream.
```

File: docs/pages/contributing.md [hunk 13]
```diff
@@ -1,58 +1,96 @@
...
+
+* Documentation on how to sync/rebase your fork can be found here:
+[https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork)
```
Comment (redhatrises) [inline docs/pages/contributing.md:69]:
```suggestion
* Documentation on how to sync/rebase your fork can be found at [https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork)
```

File: docs/pages/contributing.md [hunk 14]
```diff
@@ -1,58 +1,96 @@
...
+
+* For handling merge conflicts, refer to:
+[https://opensource.com/article/20/4/git-merge-conflict](https://opensource.com/article/20/4/git-merge-conflict)
```
Comment (redhatrises) [inline docs/pages/contributing.md:72]:
```suggestion
* For handling merge conflicts, refer to [https://opensource.com/article/20/4/git-merge-conflict](https://opensource.com/article/20/4/git-merge-conflict)
```

**Review by redhatrises — CHANGES_REQUESTED** [review]
Additionally if you can add back a section on squashing duplicate commits as it is also a noticeable problem, that would be helpful!

File: docs/pages/contributing.md (hunk 9 above)
Comment (shawndwells) [reply docs/pages/contributing.md:30]: Sure. Note GitHub automatically turns URLs into clickable links.

File: docs/pages/contributing.md [hunk 15]
```diff
@@ -1,58 +1,91 @@
 ---
-title: Contributing
-description: This is a documentation page.
+title: Contributing to a Project
```
Comment (shawndwells) [inline docs/pages/contributing.md:2]:
```suggestion
title: Contributing
```

**Comment by shawndwells** [comment]
@redhatrises this is ready for re-review

**Comment by redhatrises** [comment]
LGTM

**Pull request merged**